lib/wpc_proto/internal_modules/proto_config.c

```diff
@@ -15,11 +15,238 @@
 #include "logger.h"
 
 
-bool Proto_config_init(void)
+/** Structure to hold config from node */
+typedef struct sink_config
+{
+    app_addr_t node_address;
+    net_addr_t network_address;
+    bool CipherKeySet;
+    bool AuthenticationKeySet;
+    uint16_t stack_profile;
+    uint16_t hw_magic;
+    uint16_t ac_limit_min;
+    uint16_t ac_limit_max;
+    uint16_t app_config_max_size;
+    uint16_t version[4];
+    uint16_t ac_range_min_cur;  // 0 means unset ?
+    uint16_t ac_range_max_cur;
+    uint8_t max_mtu;
+    uint8_t ch_range_min;
+    uint8_t ch_range_max;
+    uint8_t pdu_buffer_size;
+    uint8_t StackStatus;
+    net_channel_t network_channel;
+    app_role_t app_node_role;
+} sink_config_t;
+
+/* TODO : Protect config access */
+
+/** Sink config storage */
+sink_config_t m_sink_config;
+
+/* Typedef used to avoid warning at compile time */
+typedef app_res_e (*func_1_param)(void * param);
+typedef app_res_e (*func_2_param)(void * param1, void * param2);
+
+/**
+ * \brief   Generic function to read one or two parameters from node
+ * \param   f
+ *          The function to call to get the parameters (type func_1_t or  func_2_t)
+ * \param   var1
+ *          Pointer to first parameter
+ * \param   var2
+ *          Pointer to second parameter (can be NULL)
+ * \param   var_name
+ *          Parameter name
+ * \return  True if correctly read, false otherwise
+ */
+static bool get_value_from_node(void * f,
+                                void * var1,
+                                void * var2,
+                                char * var_name)
 {
+    app_res_e res = APP_RES_INTERNAL_ERROR;
+
+    if (var2 == NULL)
+    {
+        res = ((func_1_param) f)(var1);
+    }
+    else
+    {
+        res = ((func_2_param) f)(var1, var2);
+    }
+
+    if (res != APP_RES_OK)
+    {
+        LOGE("Cannot get %s from node\n", var_name);
+        return false;
+    }
+
     return true;
 }
 
+void convert_role_to_proto_format(app_role_t role, wp_NodeRole * proto_role)
+{
+    pb_size_t flags_count = 0;
+    proto_role->role = GET_BASE_ROLE(role);
+    if (GET_ROLE_OPTIONS(role) & APP_ROLE_OPTION_LL)
+    {
+        proto_role->flags[flags_count] = wp_NodeRole_RoleFlags_LOW_LATENCY;
+        flags_count++;
+    }
+    if (GET_ROLE_OPTIONS(role) & APP_ROLE_OPTION_AUTOROLE)
+    {
+        proto_role->flags[flags_count] = wp_NodeRole_RoleFlags_AUTOROLE;
+        flags_count++;
+    }
+    proto_role->flags_count = flags_count;
+}
+
+static bool initialize_config_variables()
+{
+    _Static_assert(member_size(wp_NodeRole, flags) == (2 * sizeof(wp_NodeRole_RoleFlags)) );
```

Comment:
Not sure to understand the purpose of this test.

From proto generated file we have:
```
typedef enum _wp_NodeRole_RoleFlags {
    wp_NodeRole_RoleFlags_LOW_LATENCY = 1,
    wp_NodeRole_RoleFlags_AUTOROLE = 2
} wp_NodeRole_RoleFlags;

/* Struct definitions */
typedef struct _wp_NodeRole {
    wp_NodeRole_BaseRole role;
    pb_size_t flags_count;
    wp_NodeRole_RoleFlags flags[2];
} wp_NodeRole;
```

So you are doing a check that protoc has respected our following option hint ;-) :

`wp.NodeRole.flags max_count:2`

Reply:
Ok, but in this case, we have instructed the code to do it with "options". So I feel that the check is a bit overkill.
If we don't trust protoc code generator, then we have a big problem :-)